### PR TITLE
build: Test against Go 1.20

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+run:
+  deadline: 10m
+
+output:
```
Comment (jholdstock) [inline .golangci.yml:4]: I suggest removing this, [rationale here](https://github.com/decred/dcrvotingweb/pull/290#discussion_r1229206494)

File: go.mod [hunk 2]
```diff
@@ -3,25 +3,23 @@ module github.com/decred/dcrtime
...
 require (
-	decred.org/dcrwallet/v2 v2.0.8
-	github.com/davecgh/go-spew v1.1.1
-	github.com/decred/dcrd/certgen v1.1.1
-	github.com/decred/dcrd/chaincfg/chainhash v1.0.3
-	github.com/decred/dcrd/chaincfg/v3 v3.1.1
-	github.com/decred/dcrd/dcrutil/v3 v3.0.0
-	github.com/decred/dcrd/txscript/v3 v3.0.0
-	github.com/decred/dcrd/wire v1.5.0
+	decred.org/dcrwallet/v2 v2.2.0
```
Comment (davecgh) [inline go.mod:6]: This is v3 now.

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -9,16 +9,16 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go: [1.18, 1.19]
+        go: ['1.19', '1.20']
     steps:
       - name: Set up Go
-        uses: actions/setup-go@84cbf8094393cdc5fe1fe1671ff2647332956b1a #v3.2.1
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 #v3.5.0
```
Comment (davecgh) [inline .github/workflows/go.yml:15]: Latest is `fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.01` .

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -9,16 +9,16 @@ jobs:
...
         with:
           go-version: ${{ matrix.go }}
       - name: Check out source
-        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b #v3.0.2
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab #v3.5.2
```
Comment (davecgh) [inline .github/workflows/go.yml:19]: Latest is ` c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3` .

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -9,16 +9,16 @@ jobs:
...
       - name: Install Linters
-        run: "curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.49.0"
+        run: "curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.53.2"
```
Comment (davecgh) [inline .github/workflows/go.yml:21]: Latest is `v1.53.3`.

File: dcrtimed/backend/filesystem/filesystem_test.go [hunk 6]
```diff
@@ -53,7 +52,7 @@ func TestEncodeDecode(t *testing.T) {
 	}
 
 	if !reflect.DeepEqual(fr, *fr2) {
-		t.Fatalf("want %v got %v", spew.Sdump(fr), spew.Sdump(*fr2))
+		t.Fatalf("want %#v got %#v", fr, *fr2)
```
Comment (davecgh) [inline dcrtimed/backend/filesystem/filesystem_test.go:55]: Seems like this won't produce as useful output.

File: dcrtimed/backend/filesystem/fsck.go [hunk 7]
```diff
@@ -497,8 +513,8 @@ func (fs *FileSystem) fsckGlobal(options *backend.FsckOptions, empties map[int64
 	// Make sure we don't have any empties left over
 	if len(empties) != 0 && options.Fix {
 		// Shouldn't happen
-		return fmt.Errorf("   *** ERROR empties not pruned: %v ",
-			spew.Sdump(empties))
+		return fmt.Errorf("   *** ERROR empties not pruned: %#v ",
```
Comment (davecgh) [inline dcrtimed/backend/filesystem/fsck.go:516]: Similarly, this won't produce as nice of output.

**Review by davecgh — CHANGES_REQUESTED** [review]

**Review by davecgh — APPROVED** [review]

**Pull request merged**